x-pack/plugins/security_solution/public/common/hooks/use_navigate_to_alerts_page_with_filters.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { encode } from '@kbn/rison';
+
+import { SecurityPageName } from '../../../common/constants';
+import { formatPageFilterSearchParam } from '../../../common/utils/format_page_filter_search_param';
+import type { FilterItemObj } from '../components/filter_group/types';
+import { useNavigation } from '../lib/kibana';
+import { URL_PARAM_KEY } from './use_url_state';
+
+export const useNavigateToAlertsPageWithFilters = () => {
+  const { navigateTo } = useNavigation();
+
+  return (filterItems: FilterItemObj | FilterItemObj[]) => {
+    const url = encode(
+      formatPageFilterSearchParam(Array.isArray(filterItems) ? filterItems : [filterItems])
+    );
+
+    navigateTo({
+      deepLinkId: SecurityPageName.alerts,
+      path: `?${URL_PARAM_KEY.pageFilter}=${url}`,
```

Comment:
🤔 hmmm aren't we losing the rest of the parameters here? maybe we could get the urlSearch for the alerts page, and then use `URLSearchParams` to change only the parameter we need:

```
const urlSearch = useGetUrlSearch(SecurityPageName.alerts);
const searchParams = new URLSearchParams(urlSearch);

searchParams.set(URL_PARAM_KEY.pageFilter, format....);

navigateTo({
      deepLinkId: SecurityPageName.alerts,
      path: `?${searchParams.toString()}`,
});
```

Reply:
As discussed, this is only needed if this is a link that the user can open in a different tab or do right-click and copy the URL. Ignore this comment if the component doesn't have the `href` attr.

Reply:
Thanks Sergi, I confirmed no hrefs were used.